x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_side/final_side.tsx

```diff
@@ -5,20 +5,25 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useCallback } from 'react';
 import { EuiTitle } from '@elastic/eui';
-import type { DiffableRule } from '../../../../../../../common/api/detection_engine';
-import { FieldReadOnly } from '../final_readonly/field_readonly';
 import { SideHeader } from '../components/side_header';
 import { FinalSideHelpInfo } from './final_side_help_info';
 import * as i18n from './translations';
+import { FinalReadOnly } from '../final_readonly/final_readonly';
+import { FinalEdit } from '../final_edit/final_edit';
+import { FinalSideMode } from './constants';
+import type { UpgradeableDiffableFields } from '../../../../model/prebuilt_rule_upgrade/types';
 
 interface FinalSideProps {
-  fieldName: string;
-  finalDiffableRule: DiffableRule;
+  fieldName: UpgradeableDiffableFields;
 }
 
-export function FinalSide({ fieldName, finalDiffableRule }: FinalSideProps): JSX.Element {
+export function FinalSide({ fieldName }: FinalSideProps): JSX.Element {
+  const [mode, setMode] = React.useState<FinalSideMode>(FinalSideMode.READONLY); // TODO: Replace with state from context
+  const setReadOnlyMode = useCallback(() => setMode(FinalSideMode.READONLY), []);
+  const setEditMode = useCallback(() => setMode(FinalSideMode.EDIT), []);
```

Comment:
Do you wanna move it to a context? You could "lift up" `FieldEditContext`, move these lines there and rename it to `FieldSideContext`.

Reply:
Thanks! Fixed.